pkg/network/ebpf/c/http.h

```diff
@@ -102,6 +103,20 @@ static __always_inline void http_enqueue(http_transaction_t *http, conn_tuple_t
     }
 }
 
+static __always_inline void add_tags_http_tuple_enqueue(skb_info_t *skb_info, u64 tags, u64 started, u64 classified) {
```

Comment:
can you move this to `tls.c`?
I find both the function name and location a bit misleading. At first I thought it was a general purpose function that would add tags to any `http_transaction_t` and enqueue it, but then I realized that it is TLS specific.

aside from that, could you perhaps leave a comment about what does this `http_transaction_t` represent? It sort of looks like "synthetic" data as you're effectively hard-coding some of its params.
also, I'm a little bit worried that you're enqueuing a transaction that is missing certain fields (like the `request_method` and the `request_fragment`, which could cause this to be inadvertently dropped on the user space side.

Reply:
that was the initial idea to reuse it, but in the current http code we do that in multiple steps (that is fine)
function name come from a long way, you right I renamed it`http_enqueue_with_tags` and move it in `tls.h`

I did start a doc where I added some specifies of the TLS protocol, but I will add a comment in the code as well
https://docs.google.com/document/d/1gC5goIdSQbVe1j-DpYZC6L0wuuy4180hcN0Yr9b0MUA/edit#

Reply:
I updated this branch with 
* fixup that you ask : https://github.com/DataDog/datadog-agent/commit/4c6f7a43d2898f015f1a98c7741f0859bf516d07
* fix gnutls, we must hook gnutls_deinit() https://github.com/DataDog/datadog-agent/commit/8d6056522a0cbaba200ef227389f98e7a8007b6d
* a fix in the TLS proto : send only the HttpStats with time to first APP packet if no tag exist https://github.com/DataDog/datadog-agent/commit/c9ddb6eb11b50ef732a440209117331eb230ac8f
* curl/wget gnutls libssl tracer_test : https://github.com/DataDog/datadog-agent/commit/24185f52576fc2fe1e08787d5a87e2871c96e52c